provider/image.go

```diff
@@ -286,11 +286,24 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		}
 	}
 
+	dist, _, err := docker.ImageInspectWithRaw(ctx, img.Name)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// The repoDigest should be populated after a push. Clients may choose to throw an error or coerce
+	// this to a non-optional value.
+	repoDigest := resource.NewNullProperty()
+	if len(dist.RepoDigests) > 0 {
+		repoDigest = resource.NewStringProperty(dist.RepoDigests[0])
```

Comment:
We need to ensure that `repoDigest` always returns a string, because we declare it as a string in the schema.

Here, it will only be a string if `dist.RepoDigests` is greater than 0.